homeassistant/components/withings/common.py

```diff
@@ -280,27 +639,319 @@ async def update_sleep_summary(self) -> SleepGetSummaryResponse:
+            add_value(GetSleepSummaryField.RR_AVERAGE, data.rr_average)
+            add_value(GetSleepSummaryField.RR_MIN, data.rr_min)
+            add_value(GetSleepSummaryField.RR_MAX, data.rr_max)
+
+        # Set the value to the calculated total or None.
+        for field, value in values.items():
+            if value is not None:
+                value = round(value, 4)
+            await self._async_set_data_for_measure_type(field, value)
+
+    async def _async_set_data_for_measure_type(self, measure_type, value: Any) -> None:
+        attributes = [
+            attribute
+            for attribute in WITHINGS_ATTRIBUTES
+            if attribute.measute_type == measure_type
+        ]
+
+        for attribute in attributes:
+            await self.async_set_data(attribute, value)
+
+    async def async_set_data(self, attribute: WithingsAttribute, value: Any) -> None:
+        """Set data for a specific measurement."""
+        entity_id = await async_get_entity_id(
+            self._hass, attribute, self._profile, self._user_id
+        )
+
+        if not entity_id:
+            return
+
+        if attribute.platform == BINARY_SENSOR_DOMAIN:
+            state = STATE_ON if value is True else STATE_OFF
+        else:
+            state = value
+
+        self._hass.states.async_set(
+            entity_id=entity_id,
+            new_state=state,
+            attributes={
+                "friendly_name": attribute.friendly_name,
+                "unit_of_measurement": attribute.unit_of_measurement,
+            },
+        )
+
+    @property
+    def user_id(self) -> int:
+        """Get the user_id of the authenticated user."""
+        return self._user_id
+
+    @property
+    def profile(self) -> str:
+        """Get the profile."""
+        return self._profile
+
+    @property
+    def is_available(self) -> bool:
+        """Return is the service is available."""
+        return self._is_available
+
+    def _on_api_success(self) -> None:
+        if not self._is_available:
+            _LOGGER.info("Service is available.")
+            self.hide_auth_notification()
+        self._is_available = True
+
+    def _on_api_error(self, exception: Exception) -> None:
+        if self._is_available:
+            _LOGGER.error("Service is unavailable: %s", str(exception))
+            if isinstance(exception, NotAuthenticatedError):
+                self.show_auth_notification()
+
+        self._is_available = False
+
+    def hide_auth_notification(self) -> None:
+        """Hide the auth error notification."""
+        persistent_notification.async_dismiss(
+            hass=self._hass, notification_id=self._api_notification_id
+        )
+
+    def show_auth_notification(self) -> None:
+        """Show the auth error notification."""
+        message = f"Withings auth tokens expired for profile {self._profile}, remove and re-add the integration."
+        persistent_notification.async_create(
```

Comment:
Instead of showing this as a notification, instantiate a new config flow. If the unique IDs of the config flows are the same, it will replace the old config flow.

Reply:
This may get tricky. Withings support multiple profiles per account. Each profile represent a person's information. The Withings API does not expose Personally Identifiable information (PII). During OAuth2 setup, the user must select which profile they want from the Withings website. After which on Home Assistant (HA), we also have to ask the user which profile this applies to.
The purpose of HA asking the profile name is because we cannot get it from Withings and it makes is easy for the HA user to know which sensor belongs to which person.

So it's easy enough to start a new config flow but my question is this. On a HA setup with multiple Withings integrations setup, how does the HA user know which Withings profile needs to be re-authenticated? Can I pass the profile to the flow and have that used as a title or something?

Reply:
Yes, you can use placeholders for the title (use `flow_title` in `strings.json`) or use placeholders for the description.